lib/libzfs/os/freebsd/libzfs_compat.c

```diff
@@ -118,15 +125,31 @@ retry:		(void) execve(bp, argv, envp);
 		case ENOEXEC:
 			for (cnt = 0; argv[cnt]; ++cnt)
 				;
-			memp = alloca((cnt + 2) * sizeof (char *));
+
+			/*
+			 * cnt may be 0 above; always allocate at least
+			 * 3 entries so that we can at least fit "sh", bp, and
+			 * the NULL terminator.  We can rely on cnt to take into
+			 * account the NULL terminator in all other scenarios,
+			 * as we drop argv[0].
+			 */
+			memp = alloca(MAX(3, cnt + 2) * sizeof (char *));
```

Comment:
Should this be a MIN(3, cnt +2) since we want at least 3 entries?

Reply:
That's literally what this does, yes. `MIN` would pick either `(cnt = 0) + 2 = 2` or `3`.

Reply:
Oops, my bad. Too early.